src/TLuaInterpreter.cpp

```diff
@@ -4855,10 +4858,6 @@ int TLuaInterpreter::searchRoom(lua_State* L)
         if (!roomIdsFound.isEmpty()) {
             for (int i : roomIdsFound) {
                 TRoom* pR = host.mpMap->mpRoomDB->getRoom(i);
```

Comment:
... so that when we use the Id here we know (because of the pseudo-single threading-ness of the main Mudlet thread) that the pointer that it will return from the call to `TRoomDB::getRoomId` is still going to be valid - as nothing will have changed the contents of `TRoom::rooms` between the two places.

This is not something that automated code analysis will detect I think - so the warning from whatever the system was is a false positive in this particular case - IMHO.